docs/architecture.md

```diff
@@ -2,32 +2,76 @@
 
 ## Flux
 
 ### Architecture Pattern
 
-In short, Flux architecture design maintains a unidirectional data flow, in which a global Dispatcher receives Actions & dispatches them to appropriate Stores. The Stores, in turn, process data & provide the source of truth for the Views. As users interact with the Views, any updates are made via a dispatched Action and the cycle begins again. See this [flux architecture](https://facebook.github.io/flux/docs/overview.html) writeup for more details on the Flux architecture scheme.
+In short, Flux architecture design maintains a unidirectional data flow, in which a global Dispatcher receives Actions & dispatches them to appropriate Stores. The Stores, in turn, process data & provide the source of truth for the Views. As users interact with the Views, any updates are made via a dispatched Action and the cycle begins again. See this [flux architecture](https://facebook.github.io/flux/docs/overview.html) writeup for more details on the original Flux architecture scheme.
+
+Lockbox implements a modified version of the described architecture (LockFlux), keeping in mind that the original implementation ignores asynchronous work. The key difference is in the implementation of an `ActionHandler` class. The `ActionHandler`s in some cases are a simple pass-through class for the dispatcher, but in others do some background work before dispatching the `Action`.
+
+### Memory Management
+
+The six major components of this architecture (`View`, `Presenter`, `Store`, `Dispatcher`, `ActionHandler`, and `Action`) have distinct lifecycle management based on their functions.
+
+`View`/`Presenter` pairs are allocated and de-allocated as views get displayed or hidden in turn.
+
+`Store`s, `ActionHandler`s, and the `Dispatcher` are global singleton objects, meaning that they get lazy-loaded by the application as their shared members get accessed by the `Presenter`s for view configuration or dispatching.
+
+`Action`s get deallocated as soon as they reach the end observer for their intended function.
+
+More details about all of the major LockFlux components can be found in the following documentation.
+
+### View/Presenter
+
+All views are bound to a presenter[1]. In this separation, the presenter is responsible for all business logic, and the view is abstracted to a simple protocol. The view is responsible for UIKit-specific configuration and passing user input to its presenter for handling. This allows any complex view-related configuration to be abstracted when dealing with business logic changes, and vice versa. Presenters should never import UIKit in this separation of concerns. The `View` component of these view-presenter pairs maintains a strong reference to its `Presenter`, while the `Presenter` maintains a `weak` reference to the view to avoid retain cycles under `ARC`.
```

Comment:
To make the end-notes linkable, I would change the reference(er.g., `[1]`) to a link (e.g., `[[1](#note-1)]`) and update the end-notes accordingly (see other CR comments)